staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -682,6 +684,21 @@ func TestRoundTripSocks5AndNewConnection(t *testing.T) {
 	}
 }
 
+func TestRoundTripPassesContextToDialer(t *testing.T) {
+	urls := []string{"http://127.0.0.1:1233/", "https://127.0.0.1:1233/"}
+	for _, u := range urls {
+		t.Run(u, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			cancel()
+			req, err := http.NewRequestWithContext(ctx, "GET", u, nil)
+			require.NoError(t, err)
+			spdyTransport := NewRoundTripper(&tls.Config{})
+			_, err = spdyTransport.Dial(req)
+			assert.EqualError(t, err, "dial tcp 127.0.0.1:1233: operation was canceled")
```

Comment:
Dialer returns a `*net.OpError` but it's a generic thing. Within it there is an `Err` field which is the original error. In Go 1.18.x it's NOT possible to check it, but in the yet unreleased 1.19 it will be possible via `errors.Is()`. See https://github.com/golang/go/issues/51428 (commit https://github.com/golang/go/commit/85b5f86584686677c554b2538a7edee96d684aea). In any case, I think it's not that important as it's a test only. String comparison is probably good enough for a test.